src/workspace-data/data-table/shared/RenameColumnModal.ts

```diff
@@ -0,0 +1,135 @@
+import { Fragment, ReactNode, useState } from 'react';
+import { div, h } from 'react-hyperscript-helpers';
+import { ButtonPrimary, IdContainer, spinnerOverlay } from 'src/components/common';
+import { ValidatedInput } from 'src/components/input';
+import Modal from 'src/components/Modal';
+import { reportError } from 'src/libs/error';
+import { FormLabel } from 'src/libs/forms';
+import * as Utils from 'src/libs/utils';
+import validate from 'validate.js';
+
+export type RenameColumnModalProps = {
+  onDismiss: () => void;
+  onSuccess: () => void;
+  entityType: string;
+  attributeNames: string[];
+  oldAttributeName: string;
+  dataProvider: any;
+};
+
+export const RenameColumnModal = (props: RenameColumnModalProps): ReactNode => {
+  // State
+  const [newAttributeName, setNewAttributeName] = useState('');
+  const [isBusy, setIsBusy] = useState(false);
+  const { onDismiss, onSuccess, entityType, attributeNames, oldAttributeName, dataProvider } = props;
+
+  // For both GCP and Azure:
+  // Cannot be blank or matching an existing attribute
+  const mutualColumnNameErrors = validate.single(newAttributeName, {
+    presence: {
+      allowEmpty: false,
+      message: 'Column name is required',
+    },
+    exclusion: {
+      within: attributeNames,
+      message: "'%{value}' already exists as an attribute name",
+    },
+  });
+
+  // On GCP only, cannot be one of these reserved words:
+  //   “name”
+  //   “entityType”
+  //   “${entityType}_id”, where ${entityType} is the name of the data table
+  // A single colon is allowed (for namespacing), otherwise only alphanumeric characters, underscores, and dashes are allowed
+  const gcpColumnNameErrors =
```

Comment:
Note: this does not validate the namespace itself, so an invalid namespace will still go to the backend before coming back a failure.